lib/hammer_cli/options/option_family.rb

```diff
@@ -67,6 +74,9 @@ def child(switches, type, description, opts = {}, &block)
       end
 
       def adopt(child)
+        raise ArgumentError, 'Parent cannot be a child within the same family' if child == @parent
```

Comment:
Not sure though if this should raise an exception (to prevent bugs) or just print into debug log (for devs info).